docs/design/coreclr/botr/guide-for-porting.md

```diff
@@ -0,0 +1,440 @@
+
+1.  Defines that are architecture specific specifying the sizes of various data
+    structures the VM should create, and such
+
+2.  Defines which specify which of various jit helpers should be replaced with
+    asm functions instead of the portable C++ implementations
+
+3.  The CalleeSavedRegisters, ArgumentRegisters, and FloatArgumentRegisters as
+    needed to describe the calling convention for the platform
+
+4.  The ClrFlushInstructionCache function. If the architecture doesn’t actually
+    need to manually flush the icache, then this function is empty.
+
+5.  Various functions for decoding and manipulating jump instructions. These are
+    used by various stub routines to predict where code will go, and to produce
+    simple jump stubs.
+
+6.  The StubLinkerCpu class for the architecture. Each Architecture defines its
+    own StubLinkerCpu api surface and uses it to produce VM generated code.
+    There is a small set of apis that are called from general purpose vm code
+    (EmitComputedInstantiatingMethodStub, EmitShuffleThunkshared) across
+    multiple architectures, and then there are the individual assembly
+    instruction emission functions which are architecture specific. The
+    StubLinker is used to generate complex stubs, where the set of assembly
+    instructions emitted varies from stub to stub.
+
+7.  Various stub data structures. Many very simple stubs are not emitted via an
+    emission of a stream of bytes, but instead are exceptionally regular, and
+    are effectively the same instructions for each different stub, only with
+    slightly different data members. Instead of using the StubLinker mechanism,
+    the VM instead has structures that represent the entirety of the stub and
+    its associate data, and fill in the assembly instructions and data fields
+    with a normal constructor call setting magic numbers. In addition to being
+    executable, these stubs are often parsed to determine exactly what a given
+    function is, what it is doing, where control flow will lead to, etc.
+
+#### Virtualcallstubcpu.h
+
+This header is used to provide implementation of various stubs as used by
+virtual stub dispatch. These stubs are the lookup, resolver, and dispatch stubs
+as described in [Virtual Stub Dispatch](virtual-stub-dispatch.md). These are 
+maintained in a separate file from the rest of cgencpu.h for historical reasons,
+and for reasons of size (there is quite a lot of logic here.)
+
+System.Private.CoreLib
+----------------------
+
+### Initial Bring up
+
+In System.Private.CoreLib there is no work necessary for initial bring up.
+
+### Complete support
+
+Complete support involves changing the publicly visible api surface of the
+product. Doing so is a process handled via public issues on GitHub and
+discussions with the api review board.
+
+-   Adding support for the architecture to the
+    System.Reflection.ImageFileMachine enum, and
```

Comment:
This is not actually necessary (unless you are on Windows and need to support managed C++).

Reply:
Managed C++ doesn't actually use this either. It's really just folks who want to look at assembly names and such. Its … fairly marginal in use.

Reply:
I would just say that this optional, or just completely drop this note completely. We have not added one `ProcessorArchitecture.Arm64` even though the Arm64 port is done. It is not worth it to add because of it is not required to make anything work in .NET Core. It would be just a busy work.